Xamarin.Forms.Platform.Android/CollectionView/ItemsViewRenderer.cs

```diff
@@ -542,11 +542,13 @@ protected virtual void ReconcileFlowDirectionAndLayout()
 
 			var effectiveFlowDirection = ((IVisualElementController)Element).EffectiveFlowDirection;
 
-			if (effectiveFlowDirection.IsRightToLeft())
+			if (effectiveFlowDirection.IsRightToLeft()
```

Comment:
Can we just delete this code to fix this?

```C#
if (effectiveFlowDirection.IsRightToLeft())
			{
				linearLayoutManager.ReverseLayout = true;
			}
			else if (effectiveFlowDirection.IsLeftToRight())
			{
				linearLayoutManager.ReverseLayout = false;
			}
```

I'm not totally clear what it's accomplishing. If the flow direction on the CollectionView is set to RightToLeft then the *LayoutDirection* on the RecyclerView gets set to *LayoutDirection.RTL*

I tested with your reproduction here
https://github.com/xamarin/Xamarin.Forms/issues/7397#issuecomment-532625478

And I tested with the test procedure steps you have outlined in this PR and in both cases if we just delete the ReverseLayout code then it correctly lays out RTL


Which is all the reason why setting SupportsRTL to True was causing it to break
https://github.com/xamarin/Xamarin.Forms/issues/7397#issuecomment-532625478

Because if SupportsRTL is false than LayoutDirection doesn't do anything so when you set it to true it was then getting reversed twice. Once because LayoutDirection is set to RTL and a second time because ReverseLayout is set to True

Based on all this I feel like the fix would be to just 
- Delete the code that sets *ReverseLayout* at all
- Also modify the code inside *SingleSnapHelper.IsLayoutReversed* to test for LayoutDirection on RecyclerView

The only reason that I can think of why you'd use ReverseLayout would be if you want RTL support on the CollectionView but the user hasn't set SupportsRTL inside the Android  Manifest

But our documentation indicates that you need to set that to true in order for FlowDirection to work

https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/localization/right-to-left#android

If we wanted RTL to work when *SupportsRTL* is false (which I don't think we do) I think the code fix would be more like this

```C#
			if(!Context.HasRtlSupport())
			{
				if (effectiveFlowDirection.IsRightToLeft())
				{
					linearLayoutManager.ReverseLayout = true;
				}
				else if (effectiveFlowDirection.IsLeftToRight())
				{
					linearLayoutManager.ReverseLayout = false;
				}

			}
```

@hartez @samhouts  thoughts?

Reply:
@PureWeen Yeah, you can probably just delete that. I either was not aware of supportsRTL when I was originally writing/testing that stuff, or I was trying to get the support to work without it because we were targeting API 15 (and supportsRTL is from 17 onward, I think).

TBH, I can't remember. But I think you're right that the code is wrong, especially now that 19 is the minimum API level.

Reply:
Alright I pushed those changes and also modified the checks in the snap helper. Let me know if that all works as you'd expect it to. I ran a few tests with snap helpers and it seems to work fine when set to RTL.

@Elashi can you sign the CLA?

Reply:
> Because if _supportsRtl_ is false then _RecyclerView.LayoutDirection_ doesn't do anything, so when you set _supportsRtl_ to true it was then getting reversed twice. Once because _RecyclerView.LayoutDirection_ is set to RTL and a second time because _LinearLayoutManager.ReverseLayout_ is set to True

The PR will reduce the number of reverses by adding more restricting conditions.